### PR TITLE
Refactor Agent System into Next.js Backend

### PR DESCRIPTION
**Pull request by emrsyah**
## [Refactor] - Agent System Migration to Next.js Backend

**Type**: 🧹 Refactor  
**Version**: V.0.45
**Related Issue**: #28 

---

### 📝 Summary

Refactored the entire AI agent assistant system from a direct external API approach to use a Next.js backend implementation. This migration provides better control over the agent's behavior, enables session-based memory, custom tools, and streaming responses while removing the dependency on an external backend service.

### 🎯 Changes

**New Features**
- ✅ Added `src/lib/ai/agent.ts` - Main agent implementation with LangChain/LangGraph
- ✅ Added `src/lib/ai/types.ts` - TypeScript types for agent interfaces
- ✅ Added `src/lib/ai/tools.ts` - Custom tools for the agent (search, lookup, etc.)
- ✅ Added `src/lib/ai/retriever.ts` - Document retriever for context retrieval
- ✅ Added `src/lib/ai/session-memory.ts` - Session-based conversation memory
- ✅ Added `src/lib/ai/stream.ts` - Streaming utilities for real-time responses
- ✅ Added `src/lib/ai/prompts/system.ts` - System prompts for agent behavior
- ✅ Added `src/lib/ai/citation-audit.ts` - Citation auditing for responses

**Fixes**
- 🐛 Removed external `backend-jwt.ts` dependency - No longer needed with internal implementation
- 🐛 Simplified API routes - `src/app/api/chat/route.ts` reduced from 505 lines to minimal implementation
- 🐛 Updated environment configuration in `.env.example` with new required variables

**Improvements**
- 🧹 Refactored `/api/chat` and `/api/chat/ai` routes to use new internal agent
- 🧹 Modernized package dependencies (added new AI-related packages)
- ⚡ Improved session management with persistent memory

### 🔧 Technical Details

- Changed: 
  - `src/app/api/chat/route.ts` - Simplified to use internal agent
  - `src/app/api/chat/ai/route.ts` - Refactored to internal implementation
  - `src/lib/env.ts` - Added new environment variables for agent config
  - `src/app/layout.tsx` - Minor layout adjustment

- API changes: 
  - Replaced external backend calls with internal Next.js API routes
  - Added streaming support for chat responses
  - Session-based conversation context

- Database: 
  - None required

### 📊 Stats

- **Files changed**: 17
- **Lines added**: +1934
- **Lines removed**: -667

### 🧪 Testing

- [ ] Unit tests
- [ ] Integration tests
- [x] Manual testing - Verified agent functionality
- [ ] Edge cases covered

### 📸 Screenshots (if applicable)

N/A

### 🚦 Deployment Notes

- Environment variables needed:
  - `VOYAGE_API_KEY` - For embeddings
  - `OPENAI_API_KEY` - For LLM
  - (Check `.env.example` for complete list)
- Migration steps: None required
- Cache invalidation: N/A

### ⚠️ Breaking Changes?

- None - This is a backward-compatible refactor

### 🔗 Related Links

- Design docs: N/A
- Figma: N/A
- Related PRs: N/A

---

**Checklist**:
- [x] Code follows project style guide
- [x] Self-review completed
- [ ] Tests added/updated
- [x] Documentation updated
- [x] No new warnings generated

---

<a href="https://app.devin.ai/review/emrsyah/openta/pull/29" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: src/lib/ai/session-memory.ts [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+
+// ── Singleton ────────────────────────────────────────────────────────
+
+let _sessionMemory: SessionMemory | null = null;
+
+export function getSessionMemory(opts?: SessionMemoryOptions): SessionMemory {
+  if (!_sessionMemory) {
+    _sessionMemory = new SessionMemory(opts);
+  }
+  return _sessionMemory;
```
[the remark this answers is not in the document]
Comment (emrsyah) [reply src/lib/ai/session-memory.ts:379]: done resolved
